libraries/AP_HAL_Linux/Thread.cpp

```diff
@@ -85,6 +85,10 @@ void Thread::_poison_stack()
     void *stackp;
     uint32_t *p, *curr, *begin, *end;
 
+    if (_ctx == 0) {
```

Comment:
the only reason for this being 0 is if pthread_create() failed and if that fails it means _run_trampoline() will not be called.

what error are you seeing?

Reply:
musl's `pthread_create()` starts the thread before assigning to `_ctx`, so there is a race. On the other hand, I don't think this is the right solution. I think `pthread_self()` should be used instead of `_ctx` as I did here: https://github.com/ArduPilot/ardupilot/pull/13097/files#diff-a217fcd803cdb1603252ce9d840340deR88

Reply:
yeah... fixing that race would be a good outcome and using `ptherad_self()` like you did is a much better alternative than "some times we can't poison our stack"